src/cirrus/management/deployment.py

```diff
@@ -212,27 +213,55 @@ def get_payload_by_id(self, payload_id, output_fileobj):
 
         return s3.download_fileobj(bucket, key, output_fileobj)
 
-    def get_execution(self, arn):
+    def get_execution_arn(
+        self,
+        *,
+        arn: str | None = None,
+        payload_id: str | None = None,
+    ) -> str:
+        if arn:
+            return arn
+        if payload_id:
+            execs = self.get_payload_state(payload_id).get("executions", [])
+            try:
+                return execs[-1]
+            except IndexError as e:
+                raise NoExecutionsError(payload_id) from e
+        raise ValueError("Either arn or payload_id must be provided")
```

Comment:
order-swap-suggestion to reduce indentation: 
```suggestion
    def get_execution_arn(
        self,
        *,
        arn: str | None = None,
        payload_id: str | None = None,
    ) -> str:
        if arn:
            return arn
        if payload_id is None:
            raise ValueError("Either arn or payload_id must be provided")
        execs = self.get_payload_state(payload_id).get("executions", [])
        try:
            return execs[-1]
        except IndexError as e:
            raise NoExecutionsError(payload_id) from e
        
```